sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -365,20 +382,20 @@ AZ_NODISCARD az_result setup_url(CURL * p_curl, az_http_request const * p_reques
     size_t const extra_space_for_zero = az_span_length(AZ_SPAN_ZEROBYTE);
     size_t const url_final_size = az_span_length(p_request->_internal.url) + extra_space_for_zero;
     // allocate buffer to add \0
-    AZ_RETURN_IF_FAILED(az_span_malloc(url_final_size, &writable_buffer));
+    AZ_RETURN_IF_FAILED(_az_span_malloc(url_final_size, &writable_buffer));
```

Comment:
Do we have to allocate for CURL; can't we just use our http_response buffer instead?